operator/src/main/java/oracle/kubernetes/operator/helpers/ConfigMapHelper.java

```diff
@@ -172,6 +172,205 @@ Step updateConfigMap(Step next, V1ConfigMap existingConfigMap) {
+
+    Step createConfigMap(Step next) {
+      return new CallBuilder()
+          .createConfigMapAsync(domainNamespace, model, createCreateResponseStep(next));
+    }
+
+    abstract ResponseStep<V1ConfigMap> createCreateResponseStep(Step next);
+
+    protected boolean isCompatibleMap(V1ConfigMap existingMap) {
+      return VersionHelper.matchesResourceVersion(existingMap.getMetadata(), DOMAIN_V1)
+          && COMPARATOR.containsAll(existingMap, this.model);
+    }
+
+    void logConfigMapExists() {
+      LOGGER.fine(MessageKeys.CM_EXISTS, domainNamespace);
+    }
+  }
+
+  static FileGroupReader getScriptReader() {
+    return scriptReader;
+  }
+
+  interface ConfigMapComparator {
+    /** Returns true if the actual map contains all of the entries from the expected map. */
+    boolean containsAll(V1ConfigMap actual, V1ConfigMap expected);
+  }
+
+  static class ConfigMapComparatorImpl implements ConfigMapComparator {
+    @Override
+    public boolean containsAll(V1ConfigMap actual, V1ConfigMap expected) {
+      return actual.getData().entrySet().containsAll(expected.getData().entrySet());
+    }
+  }
+
+  /**
+   * Factory for {@link Step} that creates config map containing sit config
+   *
+   * @return Step for creating config map containing sit config
+   */
+  public static Step createSitConfigMapStep(Step next) {
+    return new SitConfigMapStep(next);
+  }
+
+  static class SitConfigMapStep extends Step {
+
+    SitConfigMapStep(Step next) {
+      super(next);
+    }
+
+    @Override
+    public NextAction apply(Packet packet) {
+      DomainPresenceInfo info = packet.getSPI(DomainPresenceInfo.class);
+      Domain domain = info.getDomain();
+
+      String result = (String) packet.remove(ProcessingConstants.DOMAIN_INTROSPECTOR_LOG_RESULT);
+      // Parse results into separate data files
+      Map<String, String> data = parseIntrospectorResult(result);
+      LOGGER.fine("================");
+      LOGGER.fine(data.toString());
+      LOGGER.fine("================");
+      String topologyYaml = data.get("topology.yaml");
+      if (topologyYaml != null) {
+        LOGGER.fine("topology.yaml: " + topologyYaml);
+        DomainTopology domainTopology = parseDomainTopologyYaml(topologyYaml);
+        WlsDomainConfig wlsDomainConfig = domainTopology.getDomain();
+        info.setScan(wlsDomainConfig);
+        info.setLastScanTime(new DateTime());
+        LOGGER.info(
+            MessageKeys.WLS_CONFIGURATION_READ,
+            (System.currentTimeMillis() - ((Long) packet.get(JobHelper.START_TIME))),
+            wlsDomainConfig);
+        SitConfigMapContext context =
+            new SitConfigMapContext(
+                this,
+                domain.getDomainUID(),
+                getOperatorNamespace(),
+                domain.getMetadata().getNamespace(),
+                data);
+
+        return doNext(context.verifyConfigMap(getNext()), packet);
+      }
+
+      // TODO: How do we handle no topology?
```

Comment:
What is the answer?

Reply:
I checked with Lenny and we need to think more about this. Will file a new JIRA to track it.

Reply:
Closing as resolved, as tracked via OWLS-70161